docs/content/dev-tools/option-provider.md

```diff
@@ -116,6 +120,65 @@ _valueTransform_ - if specified, defines the way the <u>value</u> will be transf
 
 If the JSON structure is such that the required array is nested deeper than the "root" node, you can add a "path" to the url like `http://acme.com/apis/sample.json/internal/path` The "path" is defined similar to a Sling suffix: it is the trailing part of the URL after the ".json/" extension.
 
+#### Enum class as datasource
+
+`@OptionSource` supports usage of Enum classes as data sources. Please take a look at the example below:
+
+```java
+
+@AemComponent(
+    path = "path/to/my/component",
+    title = "My AEM Component"
+)
+@Dialog
+public class MyComponent {
+    @DialogField
+    @RadioGroup(
+        buttonProvider = @OptionProvider(@OptionSource(
+            valueClass = MyEnum.class,
+            textMember = "getField",
+            valueMember = "methodName",
+            attributeMembers = {"defaultName", "anotherMethod"})
+        ))
+    String tag;
+}
+
+public enum MyEnum {
+    FIRST_OPTION("text1", "vale1", "first_attribute1", "second_attribute1"),
+    SECOND_OPTION("text2", "vale2", "first_attribute2", "second_attribute2"),
+    THIRD_OPTION("text3", "vale3", "first_attribute3", "second_attribute3");
+
+    private final String someField;
+    private final String someOtherField;
+    private final String anotherField;
+    private final String anotherSecondField;
+
+    MyEnum(String someField, String someOtherField, String anotherField, String anotherSecondField) {
+        this.someField = someField;
+        this.someOtherField = someOtherField;
+        this.anotherField = anotherField;
+        this.anotherSecondField = anotherSecondField;
+    }
+
+    public String getField() {
+        return this.someField;
+    }
+
+    public String methodName() {
+        return this.someOtherField;
+    }
+
+    public String defaultName() {
+        return this.anotherField;
+    }
+
+    public String anotherMethod() {
+        return this.anotherSecondField;
+    }
+}
+```
+As you can see from above, now we can populate `@OptionProvider` with `@OptionSource` build based on Enum class that is passed to _classValue_ (same works if we need a fallback option and use an Enum class as _fallbackClass_ property value). Working with Enums as `@OptionSource`'s, the main difference now is that _textMember_, _valueMember_, and _attributeMembers_ should accept Enum class public methods names. Such methods should return String or, otherwise the return Object value will be converted with corresponding toString() method. Other properties work the same way as with JCR node path or HTTP endpoints.
```

Comment:
"Such methods should return String or, otherwise the return Object value will be converted with corresponding toString() method."

Sounds like "such methods should return String but  -- never mind -- they don't need to" )) This phrase can be actually skipped without loss of sense